accounts/accounts/tests/test_end_to_end.py

```diff
@@ -415,6 +419,22 @@ def test_post_login(self):
                     .first()
                 self.assertEqual(db_session.end_time, 0)
 
+
+    def test_already_logged_in_redirect(self):
+        """User logs in, then reqeusts /login again and should be redirected"""
+        client = self.app.test_client()
+        client.environ_base = self.environ_base
+        form_data = {'username': 'foouser', 'password': 'thepassword'}
+        response = client.post('/login', data=form_data)
+        self.assertEqual(response.status_code, status.HTTP_303_SEE_OTHER)
+
+        next_page = 'https://arxiv.org/some_sort_of_next_page?cheeseburger=yes%20please'
```

Comment:
Is the domain ok to include in next_page, in the client.get call?

Reply:
In routes/ui.py#anonymous_only#wrapper
request.auth is None, when called by the test's client.get.
Maybe it needs to be adjusted to send cookies?

Reply:
> 

It does seem like a danger. 
It is critical to be able to go to URLs like https://mod.arxiv.org/something or services.dev.arxiv.org. 
We could restrict it to just subdomains of arxiv.org or have a regex.

Reply:
Re: request.auth is None, when called by the test's client.get. 

I checked that the cookies are sent and they seem to be there on the server. reqeust.auth gets set by code in arxiv-auth/user and I'm wondering if that is not setup in the tests.

Reply:
@bmaltzan Thanks again, you advice was helpful.